### PR TITLE
Compiling raw sierra

### LINKED ISSUE #17: Facilitate compilation of plain sierra programs
We are using in forge the `cairo_lang_sierra::program::Program` class and compiling it down to CASM with `cairo_lang_sierra_to_casm::compiler::compile`.

This binary should be able to compile that artifact as well

Reference:
https://github.com/foundry-rs/starknet-foundry/blob/master/crates/forge-runner/src/sierra_casm_runner.rs#L67

### PR DESCRIPTION
**Pull request by Arcticae**
Closes #17

### CODE REVIEW

File: tests/e2e/compile_raw.rs
Comment (Arcticae) [inline tests/e2e/compile_raw.rs:1]: We already do have 1.5.0 of sierra, can we test/add that one as well?

File: src/commands/compile_raw.rs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+use clap::Args;
+use serde::Deserialize;
+use serde_json::{json, Value};
+use std::path::PathBuf;
+
+#[derive(Args)]
+pub struct CompileRaw {
+    /// Path to the sierra json file
+    #[arg(short, long)]
+    pub sierra_input_path: PathBuf,
```
Comment (drknzz) [inline src/commands/compile_raw.rs:13]: would be nice to be able to pass this positionally

File: src/commands/compile_raw.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    /// Path to where casm json file will be saved
+    #[arg(short, long)]
```
Comment (drknzz) [inline src/commands/compile_raw.rs:16]: I think -o for output is more common

File: src/commands/compile_raw.rs (hunk 1 above)
Comment (drknzz) [reply src/commands/compile_raw.rs:13]: Just `sierra_path` maybe?

File: src/commands/compile_raw.rs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        .debug_info
+        .sierra_statement_info
+        .iter()
+        .map(|statement_debug_info| {
+            (
+                statement_debug_info.code_offset,
+                statement_debug_info.instruction_idx,
+            )
+        })
+        .collect();
```
Comment (drknzz) [inline src/commands/compile_raw.rs:49]: Let's extract to `serialize_sierra_statement_info` maybe

File: src/commands/compile_raw.rs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+#[derive(Deserialize)]
+pub struct SierraArtifact {
+    sierra_program: Program,
+}
+
+// `sierra_artifact` should be a json containing `sierra_program`
+pub fn compile(sierra_artifact: Value) -> Result<Value> {
+    let sierra_artifact: SierraArtifact = serde_json::from_value(sierra_artifact)
+        .context("Unable to deserialize Sierra. Make sure it is in a correct format")?;
```
Comment (drknzz) [inline src/commands/compile_raw.rs:28]: How does serde work for additional fields? Does `sierra_artifact` here have to be 
```
{"sierra_program": {}}
```
or could it be 
```
{"sierra_program": {}, "additional_field": {}}
```

**Review by drknzz — APPROVED** [review]
👏👏 nice

**Comment by war-in** [comment]
Usage demo:
https://github.com/foundry-rs/starknet-foundry/pull/1683

File: src/commands/compile_raw.rs (hunk 4 above)
Comment (war-in) [reply src/commands/compile_raw.rs:29]: Serde will just ignore additional fields 😌

**Review by Arcticae — COMMENTED** [review]
:shipit:

File: src/main.rs [hunk 5]
```diff
@@ -1,50 +1,76 @@
...
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    // Compile sierra of the contract
+    CompileContract(CompileContract),
+
+    // Compile sierra program (cairo_lang_sierra::program::Program)
+    CompileRaw(CompileRaw),
```
Comment (piotmag769) [inline src/main.rs:24]: It should be `CompileProgram`, I would consider bytecode or textual sierra a raw sierra

**Pull request merged**